spec/unit/system_file_spec.rb

```diff
@@ -93,6 +104,22 @@
     specify { expect(remote_dir.directory?).to be(false) }
   end
 
+  describe "#quoted_name" do
+    it "returns name with escaped single quotes" do
+      expect(single_quote_file.quoted_name).to eq(
+        "/bla/single-quote-file\\'-foo/"
+      )
```

Comment:
Maybe one example with an existing example path should be added so it is tested that paths without quotes are not changed.